app.py

```diff
@@ -81,69 +83,160 @@ def prepare_body_headers_with_data(request):
 
     return body, headers
 
-def prepare_body_headers_without_data(request):
+def should_use_data():
+    if AZURE_SEARCH_SERVICE and AZURE_SEARCH_INDEX and AZURE_SEARCH_KEY:
+        return True
+    return False
+
+def stream_with_data(body, headers, endpoint):
+    s = requests.Session()
+    response = {
+        "id": "",
+        "model": "",
+        "created": 0,
+        "object": "",
+        "choices": [{
+            "messages": []
+        }]
+    }
+    with s.post(endpoint, json=body, headers=headers, stream=True) as r:
+        for line in r.iter_lines(chunk_size=10):
+            if line:
+                lineJson = json.loads(line.lstrip(b'data:').decode('utf-8'))
+                response["id"] = lineJson["id"]
+                response["model"] = lineJson["model"]
+                response["created"] = lineJson["created"]
+                response["object"] = lineJson["object"]
+
+                role = lineJson["choices"][0]["messages"][0]["delta"].get("role")
+                if role == "tool":
+                    response["choices"][0]["messages"].append(lineJson["choices"][0]["messages"][0]["delta"])
+                elif role == "assistant": 
+                    response["choices"][0]["messages"].append({
+                        "role": "assistant",
+                        "content": ""
```

Comment:
Shouldn't content be set to `lineJson["choices"][0]["messages"][0]["delta"]["content"]`?

Reply:
the updates from the api come in order:
- tool message including role: "tool" and content
- message containing role: "assistant" and nothing else
- message containing delta: {"content":."..."} until the full assistant response is done. 

So this clause is for the second message only, it doesn't have content to add there anyway